addons/mail/controllers/attachment.py

```diff
@@ -91,23 +88,14 @@ def mail_attachment_delete(self, attachment_id, access_token=None, **kwargs):
             # Check through standard access rights/rules for internal users.
             attachment._delete_and_notify(message)
             return
-        # For non-internal users 2 cases are supported:
-        #   - Either the attachment is linked to a message: verify the request is made by the author of the message (portal user or guest).
-        #   - Either a valid access token is given: also verify the message is pending (because unfortunately in portal a token is also provided to guest for viewing others' attachments).
-        # sudo: ir.attachment: access is validated below with membership of message or access token
+        # For non-internal users the attachment can only be deleted if the correct `access_token`
+        # (limited as_author_access_token that is provided for the current share user) is sent.
```

Comment:
I think we can remove the check of message in this method too, now that the token should always be provided.

We have to double check if I'm not missing something, but I think we can simply use `_check_attachments_access` here too.